manifests/params.pp

```diff
@@ -40,7 +41,7 @@
   $ui_package_ensure     = 'latest'
   $ui_package_name       = 'consul_ui'
   $user                  = 'consul'
```

Comment:
Many of the complexity in this PR revolves around not using the `user` and `group` variables.
Can you simply set them to be `undef` in params.pp if the install method is docker instead? (then we won't need case statements throughout the codebase, and instead only have them in one place.